state/txindex/indexer.go

```diff
@@ -26,7 +27,7 @@ type TxIndexer interface {
 	Get(hash []byte) (*abci.TxResult, error)
 
 	// Search allows you to query for transactions.
-	Search(ctx context.Context, q *query.Query) ([]*abci.TxResult, error)
+	Search(ctx context.Context, q *query.Query, pagSettings ctypes.Pagination) ([]*abci.TxResult, int, error)
```

Comment:
So, we are importing rpc/core/types here, does it make sense?

Reply:
Notice that all implementations of this method already import this package, if I am not wrong. So we could move the `Pagination` type to here?